### PR TITLE
Add another path to the windows_jni.dll lookup.

### PR DESCRIPTION
**Pull request by pmuetschard**
This is to fix building android outside of the bazel repository.

To reproduce the issue (on a Windows machine, of course):
  1. Copy the examples/android folder from a bazel checkout to a new folder (keep the same folder structure - i.e. `<new folder>/examples/android`)
  2. Create a `WORKSPACE` file in `<new folder>` with the `android_sdk_repository` and `android_ndk_repository` rules as described in the example README.
  3. `bazel build examples/android/java/bazel:hello_world`
  4. Observe the failure to locate `windows_jni.dll`

### CODE REVIEW

**Comment by jin** [comment]
@ahumesky FYI

File: src/main/java/com/google/devtools/build/lib/windows/jni/WindowsJniLoader.java [hunk 1]
```diff
@@ -19,6 +19,12 @@
 
 /** Loads native code under Windows. */
 public class WindowsJniLoader {
+  private static final String[] SEARCH_PATHS = {
+    "io_bazel/src/main/native/windows/windows_jni.dll",
+    "io_bazel/external/bazel_tools/src/main/native/windows/windows_jni.dll",
+    "bazel_tools/src/main/native/windows/windows_jni.dll",
```
Comment (laszlocsomor) [inline src/main/java/com/google/devtools/build/lib/windows/jni/WindowsJniLoader.java]: I think this path alone should be enough (and then you don't to try multiple source paths).

Please test it against `//src/test/java/com/google/devtools/build/lib:windows-tests` passes on Windows. If it passes, then this PR is good to go.

Comment (pmuetschard) [reply src/main/java/com/google/devtools/build/lib/windows/jni/WindowsJniLoader.java]:
```
>bazel test src/test/java/com/google/devtools/build/lib:windows-tests
INFO: Analysed target //src/test/java/com/google/devtools/build/lib:windows-tests (0 packages loaded).
INFO: Found 1 test target...
Target //src/test/java/com/google/devtools/build/lib:windows-tests up-to-date:
  C:/users/pascal/appdata/local/temp/_bazel_pascal/ycrusu9q/execroot/io_bazel/bazel-out/x64_windows-fastbuild/bin/src/test/java/com/google/devtools/build/lib/windows-tests.jar
  C:/users/pascal/appdata/local/temp/_bazel_pascal/ycrusu9q/execroot/io_bazel/bazel-out/x64_windows-fastbuild/bin/src/test/java/com/google/devtools/build/lib/windows-tests.exe
INFO: Elapsed time: 5.896s, Critical Path: 5.10s
INFO: Build completed successfully, 4 total actions
//src/test/java/com/google/devtools/build/lib:windows-tests              PASSED in 5.0s
```

However, if I remove the first path in the list, the tests fail. The second path was added in 5d2d0728c027d0d2e51220a2dec46a76f35d2843, and so I don't think can be removed. Remember, the DLL is located via the runfiles manifest map and so the paths have to be exact matches to the key used in the manifest.

Comment (laszlocsomor) [reply src/main/java/com/google/devtools/build/lib/windows/jni/WindowsJniLoader.java]: Cool, thanks for checking!

**Comment by laszlocsomor** [comment]
Linking this PR to https://github.com/bazelbuild/bazel/issues/3832

**Pull request closed without merging**